src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -354,3 +354,18 @@ def __init__(self, col: Expression, delimiter: str, is_distinct: bool) -> None:
 
     def dependent_column_names(self) -> Optional[Set[str]]:
         return derive_dependent_columns(self.col)
+
+
+class Seq(Expression):
```

Comment:
what about functions like PERCENT_RANK? https://docs.snowflake.com/en/sql-reference/functions/percent_rank are we sure they won't have similar issues?

Reply:
It's a window function (in this PR, we make changes to all window functions and seq[*]), so it will be covered. I'll add a test case here

Reply:
Two questions:
1. Do we need to consider other [Controlled Distribution](https://docs.snowflake.com/en/sql-reference/functions-data-generation) functions?
2. If a user uses `builtin("seq8")`, or `function("seq8")`, we still have this problem. How about updating `builtin` and `function` too?

Reply:
1. no, there are random sequences, but we target at ordered seuqences.
2. Yea, but given we have seq8 here, I don't think users will use function("seq8"). Also I feel updating builtin and function is too restrictive. How about use this approach first, and when there is a user complaining about function("seq8"), then we update builtin and function?